drivers/pinctrl/pinctrl_gecko.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2022 Silicon Labs
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/pinctrl.h>
+
+int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt,
+			   uintptr_t reg)
+{
+		struct soc_gpio_pin rxpin;
+		struct soc_gpio_pin txpin;
+		USART_TypeDef *base = (USART_TypeDef *)reg;
```

Comment:
Pin control driver is USART specific, can this be made more generic? This way it could be used to assign other pin functions like the timer CC0 (PWM) output.

Reply:
@kmeinhar, Gecko pin driver supports USART only for now. We would make it generic later once this PR is merged.

Reply:
driver should be prepared to easily add more peripherals, see e.g. nRF one.